datahub/opal.py

```diff
@@ -142,13 +92,11 @@ def get_opal_row(data: dict[str, float]) -> pd.Series:  # type: ignore[type-arg]
     """
     data_array = []
 
-    for item in key_headers:
+    for item in list(opal_headers.values()):
```

Comment:
Strictly speaking, we don't need this `list` because the object returned by `.values()` is iterable. (it is still needed in the other places you've put it though).
```suggestion
    for item in opal_headers.values():
```